File: api-tester/src/main/java/com/revenuecat/apitester/java/PaymentModeAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.PaymentMode;

@SuppressWarnings({"unused", "SpellCheckingInspection"})
final class PaymentModeAPI {
    static void check(final PaymentMode paymentMode) {
        switch (paymentMode) {
            case PAY_AS_YOU_GO:
            case PAY_UP_FRONT:
            case FREE_TRIAL:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/PaymentModeAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.PaymentMode

@Suppress("unused", "UNUSED_VARIABLE")
private class PaymentModeAPI {
    fun check(paymentMode: PaymentMode) {
        when (paymentMode) {
            PaymentMode.PAY_AS_YOU_GO,
            PaymentMode.PAY_UP_FRONT,
            PaymentMode.FREE_TRIAL -> {
            }
        }.exhaustive
    }
}
